test/Microsoft.Azure.Devices.Edge.Test.Common/EdgeModule.cs

```diff
@@ -121,6 +121,37 @@ async Task WaitForStatusAsync()
             this.deviceId,
             this.iotHub.EntityPath);
 
+        public Task<string> WaitForEventsReceivedFromDeviceAsync(
```

Comment:
This method really shouldn't be a member of `EdgeModule` since it is a copy of `EdgeModule.WaitForEventsReceivedAsync`, but with all the module stuff stripped out.

You have a whole bunch of one-off logic directly in your test (or at least in that test's file), which is understandable since you're doing lots of manual steps that will go away when the feature exits preview. I'd put this method in there too, or if you want to be clean, create a `PnpDevice` class that is similar to `LeafDevice`, and put it in there.